packages/dom/src/dom.js

```diff
@@ -198,20 +112,60 @@ export function isVerticalEdge( container, isReverse ) {
 		return false;
 	}
 
-	const editableRect = container.getBoundingClientRect();
-	const buffer = lineHeight / 2;
+	// Calculate a buffer that is half the line height. In some browsers, the
+	// selection rectangle may not fill the entire height of the line, so we add
+	// 3/4 the line height to the selection rectangle to ensure that it is well
+	// over its line boundary.
+	const buffer = 3 * parseInt( lineHeight, 10 ) / 4;
+	const containerRect = container.getBoundingClientRect();
+	const verticalEdge = isReverse ?
+		containerRect.top > rangeRect.top - buffer :
+		containerRect.bottom < rangeRect.bottom + buffer;
 
-	// Too low.
-	if ( isReverse && rangeRect.top - buffer > editableRect.top ) {
+	if ( ! verticalEdge ) {
 		return false;
 	}
 
-	// Too high.
-	if ( ! isReverse && rangeRect.bottom + buffer < editableRect.bottom ) {
+	if ( isVertical ) {
+		return true;
+	}
+
```

Comment:
If I understood this trick correctly, we create a fake range positioned to the edges of the container as to create a rectangle that bounds the text: thanks to this we can detect void elements or text not stretched to the edge of the container. Is that right? Are there any more use cases? I guess this merits a comment :)

Reply:
Fixed in ccce7e0.